packages/nx-plugin/src/executors/cli/executor.ts

```diff
@@ -15,44 +15,49 @@ export default async function runAutorunExecutor(
   terminalAndExecutorOptions: AutorunCommandExecutorOptions,
   context: ExecutorContext,
 ): Promise<ExecutorOutput> {
-  const { objectToCliArgs, formatCommandStatus } = await import(
-    '@code-pushup/utils'
-  );
+  const { objectToCliArgs, formatCommandStatus, logger, stringifyError } =
+    await import('@code-pushup/utils');
   const normalizedContext = normalizeContext(context);
   const cliArgumentObject = parseAutorunExecutorOptions(
     terminalAndExecutorOptions,
     normalizedContext,
   );
-  const {
-    dryRun,
-    verbose,
-    command: cliCommand,
-    bin,
-  } = terminalAndExecutorOptions;
+  const { command: cliCommand } = terminalAndExecutorOptions;
+  const { dryRun, verbose, bin, ...restArgs } = cliArgumentObject;
   const command = bin ? `node` : 'npx';
   const positionals = [
     bin ?? '@code-pushup/cli',
     ...(cliCommand ? [cliCommand] : []),
   ];
-  const args = [...positionals, ...objectToCliArgs(cliArgumentObject)];
+  const args = [...positionals, ...objectToCliArgs(restArgs)];
+  const executorEnvVariables = {
+    ...(verbose && { CP_VERBOSE: 'true' }),
+  };
   const commandString = formatCommandStatus([command, ...args].join(' '), {
     cwd: context.cwd,
+    env: executorEnvVariables,
   });
-  if (verbose) {
-    logger.info(`Run CLI executor ${command ?? ''}`);
-    logger.info(`Command: ${commandString}`);
-  }
+
   if (dryRun) {
     logger.warn(`DryRun execution of: ${commandString}`);
   } else {
     try {
+      logger.debug(`With env vars: ${executorEnvVariables}`);
       await executeProcess({
         command,
         args,
         ...(context.cwd ? { cwd: context.cwd } : {}),
+        ...(verbose
+          ? {
+              env: {
+                ...process.env,
+                ...executorEnvVariables,
+              },
+            }
+          : {}),
```

Comment:
This logic is a little brittle. What if other properties were later added to `executorEnvVariables` in addition to `verbose`? The condition would have to be rewritten.

Something like this would be more future-proof:

```suggestion
        ...(Object.keys(executorEnvVariables).length > 0
          ? {
              env: {
                ...process.env,
                ...executorEnvVariables,
              },
            }
          : {}),
```

Although, since the `env` option uses `process.env` by default, I believe this simplified version would be equivalent:


```suggestion
        env: { ...process.env, ...executorEnvVariables },
```